scripts/publish.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Custom publishing script which wraps `npm publish` to only attempt publishing
+ * packages that have not yet been published.
+ *
+ * @packageDocumentation
+ */
+
+// @ts-check
+
+const { workspaces } = require('../package.json')
+const { glob } = require('glob')
+const path = require('node:path')
+const fs = require('node:fs/promises')
+const util = require('node:util')
+const exec = util.promisify(require('node:child_process').exec)
+const { spawn } = require('node:child_process')
+
+const ROOT = path.join(__dirname, '..')
+
+/**
+ * Invoke `npm publish`
+ * @param {string[]} pkgs - List of package names to publish
+ * @param {boolean} [dryRun] - Whether to publish in dry-run mode
+ * @returns {Promise<void>}
+ */
+async function publish(pkgs, dryRun = true) {
+  await new Promise((resolve, reject) => {
+    const args = ['publish', ...pkgs.map((name) => `--workspace=${name}`)]
+    if (dryRun) {
+      args.push('--dry-run')
+      console.info('*** DRY RUN *** DRY RUN *** DRY RUN *** DRY RUN ***')
+    }
+
+    console.info(`Running \`npm ${args.join(' ')}\``)
+
+    spawn('npm', args, {
+      cwd: ROOT,
+      stdio: 'inherit',
+      shell: true,
```

Comment:
don't love this, but we need it for nvm.  otherwise we will need to use something like `which` to find it